docs/configuration/recording_rules.md

```diff
@@ -44,22 +45,74 @@ the original expression every time it is needed. This is especially useful for
 dashboards, which need to query the same expression repeatedly every time they
 refresh.
 
-To add a new recording rule, add a line of the following syntax to your rule
-file:
+Recording and alerting rules exist in a rule group. Rules within a group are
+run sequentially at a regular interval.
 
-    <new time series name>[{<label overrides>}] = <expression to record>
+The syntax of a rule file is:
 
-Some examples:
+```yaml
+groups:
+  [ -<rule_group> ]
+```
+
+A simple example rules file would be:
+
+```yaml
+groups:
+  - name: example
+    - record: job:http_inprogress_requests:sum
+      expr: sum(http_inprogress_requests) by (job)
+```
+
+### `<rule_group>`
+```
+# The name of the group. Must be unqiue within a file.
+name: <string>
+
+# How often rules in the group are evaluated.
+[ interval: <duration> | default = global.evaulation_interval ]
+
+rules:
+  [ - <rule> ... ]
+```
+
+### `<rule>`
 
-    # Saving the per-job HTTP in-progress request count as a new set of time series:
-    job:http_inprogress_requests:sum = sum(http_inprogress_requests) by (job)
+The syntax for recording rules is:
+```
+  # The name of the time series to output to. Must be a valid metric name.
+  record: <string>
+
+  # The PromQL expression to evaluate. Every evaluation cycle this is
+  # evaluated at the current time, and the result recorded in the time
+  # series given by 'record'.
+  expr: <string>
+
+  # Labels to add or overwrite before storing the result.
+  labels:
+    [ <labelname>: <labelvalue> ]
+```
+
+The syntax for alerting rules is:
+```
+  # The name of the alert. Must be a valid metric name.
+  alert: <string>
+
+  # The PromQL expression to evaluate. Every evaluation cycle this is
+  # evaluated at the current time, and all resultant time series become
+  # pending/firing alerts.
+  expr: <string>
+
+  # Alerts are considered firing once they have been returned for this long.
+  # Alerts which have not yet fired for long enough are considered pending.
+  [ for: <duration> | default = 0s ]
 
-    # Drop or rewrite labels in the result time series:
-    new_time_series{label_to_change="new_value",label_to_drop=""} = old_time_series
+  # Labels to add or overwrite for each alert.
+  labels:
+    [ <labelname>: <tmpl_string> ]
```

Comment:
`tmpl_string` needs to be defined somewhere.

Reply:
Ah, I was thinking of the AM definition. Added it to the Prometheus config definitions.